### PR TITLE
Support init route and direct navigation in a browser

### PR DESCRIPTION
**Pull request by terrakok**
The PR adds a default behavior (if browser route customization was not set):
 - **if** the init url contains a fragment with valid route
 - **if** user enters an url fragment with valid route

**then** navigate to the route

https://github.com/user-attachments/assets/c2675371-968e-4577-b32d-60dbd64494b4

## Testing
 - Test it on a sample project
 - Write unit tests

### Features - Navigation
- Navigation via a browser address field

### CODE REVIEW

File: navigation/navigation-runtime/src/jsTest/kotlin/androidx/navigation/BrowserHistoryTest.kt [hunk 1]
```diff
@@ -50,7 +50,7 @@ class BrowserHistoryTest {
 
     @Test
     fun checkBrowserHistoryStateSynchronizedWithNavigation() = runTest {
-        cleanWindowHistory()
+        val initHistoryLength = goToBrowserRoot()
```
Comment (eymar) [inline navigation/navigation-runtime/src/jsTest/kotlin/androidx/navigation/BrowserHistoryTest.kt:55]: Could you please add the test task of this project as a dependency in `:mpp:webTest` ?

File: navigation/navigation-runtime/src/jsMain/kotlin/androidx/navigation/BrowserHistory.js.kt [hunk 2]
```diff
@@ -21,14 +21,19 @@ import org.w3c.dom.Window
 /**
  * Binds the browser window state to the given navigation controller.
  *
+ * If `getBackStackEntryRoute` is null, then:
+ *  1) if a browser url contains a destination route on a start then navigates to destination
+ *  2) if a user puts a new destination route to the browser address field then navigates to the new destination
+ *
+ * If there is a custom `getBackStackEntryRoute` implementation,
+ * then we don't have a knowledge how to parse urls to support direct navigation via browser address input
```
Comment (eymar) [inline navigation/navigation-runtime/src/jsMain/kotlin/androidx/navigation/BrowserHistory.js.kt]: Let's suggest to the readers of this KDoc that in such a case they can do it on the app side.

File: navigation/navigation-runtime/src/jsMain/kotlin/androidx/navigation/BrowserHistory.js.kt [hunk 3]
```diff
@@ -21,14 +21,19 @@ import org.w3c.dom.Window
...
+ *
  * @param navController The [NavController] instance to bind to browser window navigation.
  * @param getBackStackEntryRoute An optional function that returns the route to show for a given [NavBackStackEntry].
  */
 suspend fun Window.bindToNavigation(
```
Comment (eymar) [inline navigation/navigation-runtime/src/jsMain/kotlin/androidx/navigation/BrowserHistory.js.kt:34]: Did you want to mark it as Experimental?

Comment (eymar) [inline navigation/navigation-runtime/src/jsMain/kotlin/androidx/navigation/BrowserHistory.js.kt:45]: Did you want to mark it as Experimental?

**Review by eymar — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Comment by egorikftp** [comment]
@terrakok What if user open url in following format `http://localhost:8080/SomePage` ?

**Comment by terrakok** [comment]
It is called a deep link and should be handled as a deep link.

https://developer.android.com/guide/navigation/design/deep-link

**Comment by egorikftp** [comment]
I mean, how it should work in wasm. Maybe you can share sample

**Comment by terrakok** [comment]
It should be handled on a backend

**Pull request merged**